pkg/pods/pod.go

```diff
@@ -738,3 +749,42 @@ func (p *Pod) logLaunchableWarning(serviceID string, err error, message string)
 func (p *Pod) logInfo(message string) {
 	p.logger.WithFields(logrus.Fields{}).Info(message)
 }
+
+// Runs function f and emits warnings if it hasn't completed within 1 minute, 2
+// minutes, 5 minutes, 10 minutes, and each 5 minute increment afterward. This
+// is useful for identifying the cause of a very long install time, for
+// instance a disable or post-activate script hanging forever.
+func (p *Pod) withTimeWarnings(scriptType string, serviceID string, f func()) {
```

Comment:
I agree, but weakly. The pod state is only interrogated to populate the logger. This would be a useful timeutil helper if it took a logger as an argument.